lib/improver/blending/weighted_blend.py

```diff
@@ -39,6 +39,81 @@
 
 from improver.utilities.cube_manipulation import add_renamed_cell_method
 from improver.utilities.cube_checker import find_percentile_coordinate
+from improver.utilities.temporal import (
+    cycletime_to_number, find_required_lead_times)
+
+
+def conform_metadata(
+        cube, cube_orig, cycletime=None):
+    """Ensure that the metadata conforms after blending together across
+    the chosen coordinate.
+
+    Args:
+        cube (iris.cube.Cube):
+            Cube containing the metadata to be adjusted.
+        cube_orig (iris.cube.Cube):
+            Cube containing metadata that may be useful for adjusting
+            metadata on the `cube` variable.
+
+    Keyword Args:
+        cycletime (str):
+            The cycletime in a YYYYMMDDTHHMMZ format e.g. 20171122T0100Z.
+
+    Returns:
+        cube (iris.cube.Cube):
+            Cube containing the adjusted metadata.
+
+    """
+    if cube.coords("forecast_reference_time"):
+        if cycletime is None:
+            new_cycletime = (
+                np.max(cube_orig.coord("forecast_reference_time").points))
+        else:
+            cycletime_units = (
+                cube_orig.coord("forecast_reference_time").units.origin)
+            cycletime_calendar = (
+                cube.coord("forecast_reference_time").units.calendar)
+            new_cycletime = cycletime_to_number(
+                cycletime, time_unit=cycletime_units,
+                calendar=cycletime_calendar)
+        cube.coord("forecast_reference_time").points = new_cycletime
+        cube.coord("forecast_reference_time").bounds = None
+
+    if cube.coords("forecast_period"):
+        if cycletime is None:
+            forecast_period = np.min(cube_orig.coord("forecast_period").points)
```

Comment:
Is there a check anywhere before we get here that the cubes are appropriate to blend (same validity time), e.g.
`FRT = 1500Z; Validity time = 2100Z`
`FRT = 2100Z; Validity time = 2300Z`
`CUBES INCOMPATIBLE`

The merging step in your notebook gives a cube with a time coordinate containing two times.
The weighted blend step results in a cube with a validity time of 2200Z, a forecast period of 7200 and a FRT of 21Z.

`Scalar coordinates:
          forecast_period: 7200 seconds
          forecast_reference_time: 2017-10-22 21:00:00
          height: 1.5 m
          time: 2017-10-22 22:00:00`

The discrepancy between FRT and VT could easily be missed. My fear is that without a check somewhere, taking a max FRT and a min VT  could result in cubes being merged for different VTs but that information being thrown away and (nigh) impossible to recover. I'd be happy to be told I've missed something obvious here.

Reply:
I've added a check, so that an error is raised, if we're trying to blend using the forecast_reference_time or forecast_period coordinate, and the points within the time coordinate are different.

Reply:
:white_check_mark: